File: sample/src/main/java/ru/terrakok/cicerone/sample/SampleApplication.java
package ru.terrakok.cicerone.sample;

import android.app.Application;

import ru.terrakok.cicerone.sample.dagger.AppComponent;
import ru.terrakok.cicerone.sample.dagger.DaggerAppComponent;

/**
 * Created by Konstantin Tckhovrebov (aka @terrakok)
 * on 11.10.16
 */

public class SampleApplication extends Application {
    public static SampleApplication INSTANCE;
    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        INSTANCE = this;
    }

    public AppComponent getAppComponent() {
        if (appComponent == null) {
            appComponent = DaggerAppComponent.builder().build();
        }
        return appComponent;
    }
}

File: sample/src/main/java/ru/terrakok/cicerone/sample/SampleApplication.kt
package ru.terrakok.cicerone.sample

import android.app.Application
import ru.terrakok.cicerone.sample.dagger.AppComponent
import ru.terrakok.cicerone.sample.dagger.DaggerAppComponent

/**
 * Created by Konstantin Tckhovrebov (aka @terrakok)
 * on 11.10.16
 */
class SampleApplication : Application() {

    val appComponent: AppComponent by lazy {
        DaggerAppComponent.builder().build()
    }

    override fun onCreate() {
        super.onCreate()
        INSTANCE = this
    }

    companion object {
        lateinit var INSTANCE: SampleApplication
    }
}